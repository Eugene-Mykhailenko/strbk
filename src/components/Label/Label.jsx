/* eslint-disable jsx-a11y/label-has-for */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './Label.scss';

const Label = (props) => {
  const { text, isRequired, hasFocus, hasError } = props;

  return (
    <label
      className={classNames(styles.label, {
        [styles['is-required']]: isRequired,
        [styles['has-focus']]: hasFocus,
        [styles['has-error']]: hasError,
      })}
    >
      {text}:
    </label>
  );
};

Label.propTypes = {
  /** Text inside element. */
  text: PropTypes.string.isRequired,
  /** Determines whether to add a required field sign or not. */
  isRequired: PropTypes.bool,
  /** Selects element if it is currently focused. */
  hasFocus: PropTypes.bool,
  /** Sets UI to erroneous. */
  hasError: PropTypes.bool,
};

Label.defaultProps = {
  isRequired: false,
  hasFocus: false,
  hasError: false,
};

Label.styles = styles;

export default Label;
