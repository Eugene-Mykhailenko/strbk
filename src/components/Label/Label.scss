@import '../../assets/styles/abstracts/variables';

.label {
  display: inline-block;
  margin-bottom: 5px;
  font-size: 12px;
  line-height: 12px;
  color: var(--color_5);

  &.has-focus {
    color: var(--color_1);
  }

  &.has-error {
    color: var(--color_10);
  }

  &.is-required {
    &::before {
      content: '*';
      position: absolute;
      top: 0;
      right: -10px;
      font-size: 14px;
      color: var(--color_1);
    }
  }
}
