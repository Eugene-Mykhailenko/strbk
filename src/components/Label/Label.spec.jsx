import React from 'react';
import Label from './Label';

const config = {
  isRequired: true,
  hasFocus: true,
  hasError: true,
  text: 'Lorem',
};

describe('Label', () => {
  it('should render provided text', () => {
    const label = shallow(<Label text={config.text} />);

    expect(label.text()).toEqual(`${config.text}:`);
    expect(label).toMatchSnapshot();
  });

  it('should be able to indicate required status', () => {
    const label = shallow(
      <Label isRequired={config.isRequired} text={config.text} />,
    );

    expect(label.prop('className')).toMatch(/is-required/);
    expect(label).toMatchSnapshot();
  });

  it('should be able to indicate focused state', () => {
    const label = shallow(
      <Label hasFocus={config.hasFocus} text={config.text} />,
    );

    expect(label.prop('className')).toMatch(/has-focus/);
    expect(label).toMatchSnapshot();
  });

  it('should be able to indicate error status', () => {
    const label = shallow(
      <Label hasError={config.hasError} text={config.text} />,
    );

    expect(label.prop('className')).toMatch(/has-error/);
    expect(label).toMatchSnapshot();
  });
});
