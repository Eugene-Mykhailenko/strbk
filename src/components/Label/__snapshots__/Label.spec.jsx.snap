// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should be able to indicate error status 1`] = `
<label
  className="label has-error"
>
  Lorem
  :
</label>
`;

exports[`Label should be able to indicate focused state 1`] = `
<label
  className="label has-focus"
>
  Lorem
  :
</label>
`;

exports[`Label should be able to indicate required status 1`] = `
<label
  className="label is-required"
>
  Lorem
  :
</label>
`;

exports[`Label should render provided text 1`] = `
<label
  className="label"
>
  Lorem
  :
</label>
`;
